internal/queueutils/pool.go

```diff
@@ -51,6 +52,22 @@ func (p *OperationPool) SetTenants(tenants []*dbsqlc.Tenant) {
 	})
 }
 
+func (p *OperationPool) SetPartitions(partitions []int32) {
+	partitionMap := make(map[string]bool)
+
+	for _, partitionId := range partitions {
+		partitionMap[strconv.Itoa(int(partitionId))] = true
+	}
+
+	p.ops.Range(func(key, value interface{}) bool {
+		if _, ok := partitionMap[key.(string)]; !ok {
+			p.ops.Delete(key)
+		}
+
+		return true
+	})
+}
```

Comment:
not sure what the right way to do this is, so added this for now as a hack

Reply:
well so for our other jobs, we have this same thing but with tenants (`SetTenants`), and I wasn't sure how to spawn N jobs, one per partition, where we _aren't_ doing it by tenant, so I added this as a hack. Not sure what the correct way to do this is, though, or if we even need it at all. But yeah basically was just looking for a way to pass the partition id into `runProcessPayloadWAL`